src/api/aux.cpp

```diff
@@ -19,72 +19,75 @@ void parse_json_file(std::vector<Token> &tokens, std::vector<Order> &orders, std
+        if (!tkn.value()["external_price"].is_null()) {
+            std::string external_price_string = (tkn.value()["external_price"]).dump();
+            external_price = static_cast<boost::multiprecision::mpf_float>(external_price_string);
         }
-        if (i.value()["alias"].is_null())
-            tokens.push_back(Token(i.key(), "_NULL", i.value()["decimals"], t));       // TODO: Check if multiple tokens with "null" alias can create any issue
         else
-            tokens.push_back(Token(i.key(), i.value()["alias"], i.value()["decimals"], t));
-        _idx_tokens[i.key()] = _num_tokens-1;
+            external_price = -1;
+
+        if (!tkn.value()["alias"].is_null())
+            tokens.push_back(Token(tkn.key(), tkn.value()["alias"], tkn.value()["decimals"], external_price, normalize_priority));
+        else
+            tokens.push_back(Token(tkn.key(), "_NULL", tkn.value()["decimals"], external_price, normalize_priority));       // TODO: Check if multiple tokens with "null" alias can create any issue
+            
+        _idx_tokens[tkn.key()] = _num_tokens-1;
     }
 
     // For-loop that reads all ORDERS information
-    for (auto &i: json_file["orders"].items()) {
-        int t1 = std::stoi(i.key());
-        int t2 = _idx_tokens.at(i.value()["sell_token"]);
-        int t3 = _idx_tokens.at(i.value()["buy_token"]);
-        bool t4 = i.value()["is_sell_order"];
-        bool t5 = i.value()["is_liquidity_order"];
-        bool t6 = i.value()["allow_partial_fill"];
-        std::string tt7 = i.value()["sell_amount"];
-        boost::multiprecision::mpz_int t7 = static_cast<boost::multiprecision::mpz_int>(tt7);
-        std::string tt8 = i.value()["buy_amount"];
-        boost::multiprecision::mpz_int t8 = static_cast<boost::multiprecision::mpz_int>(tt8);
-        int t9 = _idx_tokens.at(i.value()["fee"]["token"]);
-        std::string tt10 = i.value()["fee"]["amount"];
-        boost::multiprecision::mpz_int t10 = static_cast<boost::multiprecision::mpz_int>(tt10);
-        int t11 = _idx_tokens.at(i.value()["cost"]["token"]);
-        std::string tt12 = i.value()["cost"]["amount"];
-        boost::multiprecision::mpz_int t12 = static_cast<boost::multiprecision::mpz_int>(tt12);
-
-        orders.push_back(Order(t1,t2,t3,t4,t5,t6,t7,t8,t9,t10,t11,t12));
+    for (auto &o: json_file["orders"].items()) {
+        int key = std::stoi(o.key());
+        int sell_token = _idx_tokens.at(o.value()["sell_token"]);
+        int buy_token = _idx_tokens.at(o.value()["buy_token"]);
+        bool is_sell_order = o.value()["is_sell_order"];
+        bool is_liquidity_order = o.value()["is_liquidity_order"];
+        bool allow_partial_fill = o.value()["allow_partial_fill"];
+        std::string sell_amount_string = o.value()["sell_amount"];
+        boost::multiprecision::mpz_int sell_amount = static_cast<boost::multiprecision::mpz_int>(sell_amount_string);
+        std::string buy_amount_string = o.value()["buy_amount"];
```

Comment:
This pattern is used in so many places, that you could consider to factor it out into a `parse_int_value(std::string const& key)` function.